src/control/actor/host/notify.go

```diff
@@ -57,27 +53,41 @@ func (c *HostNotifyCmd) listenCloseF(net network.Network, addr ma.Multiaddr) {
 
 func (c *HostNotifyCmd) connectedF(net network.Network, conn network.Conn) {
 	logrus.Info("connection detected: ", conn.RemotePeer().String())
-
-	// request metadata as soon as we connect to a peer
-	peerData, err := PollPeerMetadata(conn.RemotePeer(), c.Base, c.PeerMetadataState, c.Store, c.PeerStore)
+	h, _ := c.Host()
+	// Request the Host Metadata
+	hInfo := ReqHostInfo(context.Background(), h, conn)
+	// Request the BeaconMetadata
+	bMetadata, err := ReqBeaconMetadata(context.Background(), h, conn.RemotePeer())
+	if err != nil {
+		log.Warn(err)
+	}
+	// request BeaconStatus metadata as we connect to a peer
+	bStatus, err := ReqBeaconStatus(context.Background(), h, conn.RemotePeer())
+	if err != nil {
+		log.Warn(err)
+	}
 	var peer metrics.Peer
-	if err == nil {
-		peer = fetchPeerExtraInfo(peerData)
-		_ = peer
-	} else {
-		log.Info("could not get metadata for peer: ", conn.RemotePeer().String(), " err: ", err)
-		// TODO: Add also IP taken from ENR of discovered peers
-		peer = metrics.Peer {
-			PeerId: conn.RemotePeer().String(),
-		}
+	// Read ENR of the Peer from the generated enode
+	n := c.Store.LatestENR(conn.RemotePeer())
+	if n.ID().String() == "" {
+		// TODO: If the peer wasn't discovered via dv5 "n" will be empty
+		// we could compose new ENR with the beacon metadata and the host info
+		log.Warn("Peer ENR not found")
 	}
+	hInfo.NodeID = n.ID().String()
```

Comment:
as you are saying above, "If the peer wasn't discovered via dv5 "n" will be empty". here you are storing it regardless. perhaps move this check inside the `if`?
but not sure i understand it: is there any other way of getting a peer that is not via dv5?

Reply:
you are right, the assignment of the nodeID should be only done if it was in our dv5 database. 
It can be possible (remote possibility but still feasible) that a peer gets connected for its initiative when we didn't receive info about it in dv5 before. 
I'm not aware of any method to obtain the nodeID without the ENR if it's not by having the privkey. Therefore, we would have to wait until we discover that peer in dv5 to later obtain the nodeID and update it. 
I would like to know your thoughts about this, seems like a very rare case though.